apps/backend/api/models/Group.js

```diff
@@ -1002,6 +1012,22 @@ module.exports = bookshelf.Model.extend(merge({
           }
         }
 
+        // Handle track case
+        if (track && track.published_at) {
+          // Only add tracks widget if it is published
+          const tracksWidget = widgets.find(w => w.get('view') === 'tracks')
+          if (tracksWidget && !tracksWidget.get('auto_added')) {
```

Comment:
There is a little trap here: If an admin adds the tracks widget to the menu before they create a track, then auto_added will be switched to true. There is then no condition where the admin visibility will ever be switched away from admin, meaning regular users will never see the tracks widget.

Probably best to separate out these checks/operations and make sure they can happen independent of one another

Reply:
ooh, interesting